common/oai/loaders/zenodo.py

```diff
@@ -0,0 +1,143 @@
+                        "oai": {
+                            "metadata": (
+                                record["metadata"] if "metadata" in record else {}
+                            ),
+                            "datestamp": record["updated"],
+                            "deleted": False,
+                            "identifier": f"oai:zenodo.org:{record['id']}",
+                            "setSpecs": self.base_query,
+                            "oai_url": self.source,
+                        },
+                        "oai_run": self.oai_run,
+                        "oai_harvester_id": self.oai_harvester_id,
+                        "manual": self.manual,
+                    },
+                )
+
+    def search_zenodo(self, params=None, url=None):
+        """
+        Sends api request to zenodo either with specific params or to specific url
+        Handles http error and returns json response
+        """
+        try:
+            if url is None:
+                response = url_get(self.source, params=params)
+            else:
+                response = requests.get(url)  # specific url to send api request
+            response.raise_for_status()
+            return response.json()
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+            return None
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+            return None
+
+    def get_all_results_zenodo(self, query) -> list[dict]:
+        """
+        Retrieves all results from zenodo using query
+        """
+        if self.start_from is not None:
+            try:
+                datetime.strptime(self.start_from, "%Y-%m-%d")
+                query = f"({query}) AND updated:[{self.start_from} TO *]"
+            except ValueError:
+                print("Start from date is not in correct format")
+                yield None
+
+        search_results = self.search_zenodo(params={"q": query, "sort": "oldest"})
+
+        while True:
+            if search_results is None:
+                yield None
+                return
+
+            for search_result in search_results.get("hits", {}).get("hits", []):
+                yield search_result
+            if search_results.get("links", {}).get("next") is not None:
+                search_results = self.search_zenodo(
+                    url=search_results.get("links").get("next")
+                )
+            else:
+                break
+
+
+if __name__ == "__main__":
+
+    def run():
+        loader = ZenodoLoader(
+            source="https://zenodo.org/api/records/",
+            all_records=True,
+            identifiers=None,
+            oai_config={
+                "setspecs": 'creators.affiliation:("České vysoké učení technické v Praze" OR "ČVUT" OR "CESKE VYSOKE UCENI TECHNICKE V PRAZE" OR "CVUT" OR "Czech Technical University In Prague" OR "CTU in Prague" OR "CTU Prague")  AND resource_type.type:dataset',
```

Comment:
Do we want only the CVUT datasets harvested from zenodo? Aren't there other institutions to be harvested?